CondCore/AlignmentPlugins/plugins/TrackerAlignmentErrorExtended_PayloadInspector.cc

```diff
@@ -66,7 +66,7 @@ namespace {
               500.0) {
       Base::setSingleIov(true);
     }
-
+    using cond::payloadInspector::Histogram1D<AlignmentErrorsExtended>::fill;
```

Comment:
I don't understand the pattern here, why is it needed here and for instance not at:
https://github.com/cms-sw/cmssw/blob/f039d3d393d17765248bc5ec0502738bce223da7/CondCore/AlignmentPlugins/plugins/TrackerAlignmentErrorExtended_PayloadInspector.cc#L118

Reply:
No idea why compiler did not complain about few other classes in this file. May be because Histogram1D explicitly implements `bool fill()` ( https://github.com/cms-sw/cmssw/blob/master/CondCore/Utilities/interface/PayloadInspector.h#L724 ) while `PlotImage` does not ( https://github.com/cms-sw/cmssw/blob/master/CondCore/Utilities/interface/PayloadInspector.h#L829 )

Reply:
but e.g. this other plot in which you did add the `using ...` is inheriting from `PlotImage`:
https://github.com/cms-sw/cmssw/pull/29789/files#diff-25914482e3ee9c4f2fcefeac3d8dfd30R96.
Said in other way, I need to implement more of these plots, how do I decide where it is appropriate? How does one reproduce the warning?

Reply:
you can create clang base IB dev rea e.g. `scram p CMSSW_11_1_CLANG_X_2020-05-10-2300` and then checkout the packages and run `scram build`. I think the reason is that few classes in https://github.com/cms-sw/cmssw/blob/master/CondCore/Utilities/interface/PayloadInspector.h do not have 
```
virtual bool fill(const std::vector<std::tuple<cond::Time_t, cond::Hash>>& iovs) { return false; }
```
so may be proper fix would be to add these in base classes ?

Reply:
@mmusich 
The final goal of the changes triggering the CLANG compiler errors is to remove the fill(vector<...>) method in favour of the fill() method. That should be the guideline for new implementations. Concerning this PR, if we are happy I'd propose to approuve it.

Reply:
@mmusich , yes looks like in few places the `using ` for `PlotImage` not needed. I will update the PR

Reply:
PR is updated now to only fix those classes which are inherited from using `cond::payloadInspector::Histogram*`

Reply:
@smuzaffar, I had a deeper look to the base class and the method 
```
virtual bool fill(const std::vector<std::tuple<cond::Time_t, cond::Hash>>& iovs) { return false; }
```
has totally disappeared from `cond::payloadInspector::Histogram*`. 
So the proper fix would be to migrate all the derive classes to use instead 
https://github.com/cms-sw/cmssw/blob/29634922c859dc7a7cec988de55ee4ff48becb6c/CondCore/Utilities/interface/PayloadInspector.h#L724-L734

at the moment all of these plots are not functional !
See 
   * normal behaviour in [CMSSW_11_1_0_pre7](https://cms-conddb.cern.ch/cmsDbBrowser/payload_inspector/Prod/CMSSW_11_1_0_pre7/eyJwbG90cyI6MSwicGxvdDEiOnsidGFnIjoiVHJhY2tlckFsaWdubWVudEVycm9yRXh0ZW5kZWRfUnVuMlN0YXJ0dXBfdjBfbWMiLCJwbG90IjoicGxvdF9UcmFja2VyQWxpZ25tZW50RXJyb3JFeHRlbmRlZFhYVmFsdWUiLCJwbHVnaW4iOiJwbHVnaW5UcmFja2VyQWxpZ25tZW50RXJyb3JFeHRlbmRlZF9QYXlsb2FkSW5zcGVjdG9yIiwidHlwZSI6Ikhpc3RvMUQiLCJpb3ZzIjp7InN0YXJ0X2lvdiI6IjEiLCJlbmRfaW92IjoiMSJ9fX0=) (working)
![image](https://user-images.githubusercontent.com/5082376/81673999-d0d0ed80-944c-11ea-8a7a-3f0d918f4c66.png)

   * not functional behaviour in the [latest IB](https://cms-conddb.cern.ch/cmsDbBrowser/payload_inspector/Prod/CMSSW_11_1_X_2020-05-11-2300/eyJwbG90cyI6MSwicGxvdDEiOnsidGFnIjoiVHJhY2tlckFsaWdubWVudEVycm9yRXh0ZW5kZWRfUnVuMlN0YXJ0dXBfdjBfbWMiLCJwbG90IjoicGxvdF9UcmFja2VyQWxpZ25tZW50RXJyb3JFeHRlbmRlZFhYVmFsdWUiLCJwbHVnaW4iOiJwbHVnaW5UcmFja2VyQWxpZ25tZW50RXJyb3JFeHRlbmRlZF9QYXlsb2FkSW5zcGVjdG9yIiwidHlwZSI6Ikhpc3RvMUQiLCJpb3ZzIjp7InN0YXJ0X2lvdiI6IjEiLCJlbmRfaW92IjoiMSJ9fX0=): 
![image](https://user-images.githubusercontent.com/5082376/81674028-dcbcaf80-944c-11ea-9caa-a42303aaa974.png)

@ggovi

Reply:
thanks @mmusich for looking in to this. Can you or @ggovi please provide a proper fix (in a new PR)?

Reply:
@smuzaffar yes, I'll prepare a fix.

Reply:
For the record I've prepared PR https://github.com/cms-sw/cmssw/pull/29817